internal/controller/processor_test.go

```diff
@@ -575,574 +817,1025 @@ var _ = Describe("AppConfiguationProvider processor", func() {
+					FeatureFlagETags: map[acpv1.Selector][]*azcore.ETag{
+						{
+							KeyFilter: &wildcard,
+						}: {
+							&fakeEtag,
+						},
+					},
+					KeyValueETags:            existingKeyValueEtags,
+					ExistingSecretReferences: cachedSecretReferences,
+					Generation:               1,
+					ConfigMapResourceVersion: &fakeResourceVersion,
+				},
+				CurrentTime:    metav1.NewTime(nowTime.Time.Add(2 * time.Second)),
+				RefreshOptions: &RefreshOptions{},
+			}
+
+			mockConfigurationSettings.EXPECT().CheckPageETags(gomock.Any(), gomock.Any()).Return(true, nil)
+			mockConfigurationSettings.EXPECT().RefreshFeatureFlagSettings(gomock.Any(), gomock.Any()).Return(allSettingsReturnedByFeatureFlagRefresh, nil)
+			mockConfigurationSettings.EXPECT().CheckPageETags(gomock.Any(), gomock.Any()).Return(true, nil)
+			mockConfigurationSettings.EXPECT().RefreshKeyValueSettings(gomock.Any(), gomock.Any(), gomock.Any()).Return(allSettingsReturnedByKeyValueRefresh, nil)
+			expectedNextKeyValueRefreshReconcileTime := metav1.NewTime(processor.CurrentTime.Time.Add(70 * time.Second))
+			expectedNextFeatureFlagRefreshReconcileTime := metav1.NewTime(processor.CurrentTime.Time.Add(1 * time.Minute))
+			cachedNextSecretReferenceRefreshReconcileTime := processor.ReconciliationState.NextSecretReferenceRefreshReconcileTime
+
+			_ = processor.PopulateSettings(&corev1.ConfigMap{ObjectMeta: metav1.ObjectMeta{
+				ResourceVersion: fakeResourceVersion,
+			}}, existingSecrets)
+
+			_, _ = processor.Finish()
+
+			// SecretsMetadata keeps the same when no changes in KVR
+			Expect(*processor.ReconciliationState.ExistingSecretReferences[secretName].SecretsMetadata["testSecretKey"].SecretId).Should(
+				Equal(*allSettingsReturnedByKeyValueRefresh.SecretReferences[secretName].SecretsMetadata["testSecretKey"].SecretId))
+			Expect(processor.ReconciliationState.KeyValueETags[acpv1.Selector{
+				KeyFilter: &testKey,
+			}]).Should(Equal(updatedKeyValueEtags[acpv1.Selector{
+				KeyFilter: &testKey,
+			}]))
+			Expect(processor.ReconciliationState.FeatureFlagETags[acpv1.Selector{
+				KeyFilter: &wildcard,
+			}]).Should(Equal(updatedFeatureFlagEtags[acpv1.Selector{
+				KeyFilter: &wildcard,
+			}]))
+			Expect(processor.ReconciliationState.NextKeyValueRefreshReconcileTime).Should(Equal(expectedNextKeyValueRefreshReconcileTime))
+			Expect(processor.ReconciliationState.NextSecretReferenceRefreshReconcileTime).Should(Equal(cachedNextSecretReferenceRefreshReconcileTime))
+			Expect(processor.ReconciliationState.NextFeatureFlagRefreshReconcileTime).Should(Equal(expectedNextFeatureFlagRefreshReconcileTime))
+		})
+
+		// when reconcile is triggered by single component refresh
+		// only update the corresponding component reconcile state and next refresh time
+		It("Should update reconcile state when keyValue pageEtag, featureFlag pageEtag and secret references updated when configuration, secret and feature flag refresh enabled", func() {
```

Comment:
Should update the corresponding target's reconcile state, and those that shouldn't be updated should be as is.